### DISCUSSION TITLE
LobCreator vs LobHelper

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#4592] [Design Proposals]
I'm wondering why we have these two interfaces fighting over the right to create blobs.

There must be some history here, and I guess `LobCreator` is supposed to be somehow more low-level. But that's right could we at least remove the `Hibernate.getLobCreator()` methods, since that is public API, and it looks like the client is supposed to be using `LobHelper`.

And it's weird to have a static method that just delegates back to its argument to call a getter method.

**Comment by gavinking** [comment]
I mean, quite apart from being confusing, the concrete problem is that it leaks internal API (`SessionImplementor`) onto the signature of a public API in org.hibernate. That ain't right.

**Comment by sebersole** [comment]
So what is the "proposal" here?

**Reply by gavinking** [reply]
Well, remove the `getLobCreator()` methods from `Hibernate` to begin with.

**Reply by gavinking** [reply]
Or, at least, remove the ones which expose internal APIs.

**Reply by gavinking** [reply]
FTR https://github.com/hibernate/hibernate-orm/pull/4595

**Comment by sebersole** [comment]
The history is a bad decision I think in retrospect.

**Reply by gavinking** [reply]
I mean, I think it's fine to have things in two layers, just that I think we should be clear on which layer is meant for *users*.

**Reply by gavinking** [reply]
(I think that's `LobHelper`, right?)
